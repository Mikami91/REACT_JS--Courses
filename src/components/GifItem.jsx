import PropTypes from 'prop-types';

const GifItem = ({ url, title }) => {
  return (
    <>
      <img src={url} alt={title} className='img' />
      <p>{title}</p>
    </>
  );
};

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default GifItem;
