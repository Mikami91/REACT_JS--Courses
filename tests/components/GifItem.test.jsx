import { render, screen } from '@testing-library/react';
// Components
import GifItem from '../../src/components/GifItem';

describe('TEST: <GifItem />', () => {
  const title = 'Dark Souls';
  const url = 'https://example.com/';

  test('Should make match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Should show gif with URL and ALT set it', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Should show the title in the component ', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
