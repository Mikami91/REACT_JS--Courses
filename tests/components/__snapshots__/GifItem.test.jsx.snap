// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST: <GifItem /> Should make match with snapshot 1`] = `
<div>
  <img
    alt="Dark Souls"
    class="img"
    src="https://example.com/"
  />
  <p>
    Dark Souls
  </p>
</div>
`;
